### PR TITLE
Measure voltage supplied to chip

### PR DESCRIPTION
**Pull request by dhalbert**
Add `microprocessor.cpu.voltage` to measure the regulated voltage supplied to the chip. Monitoring of this value can detect when the regulated voltage is slumping due to a low battery.

This is typically 3.3V. Note that this does not measure the actual on-chip voltage used to urn the CPU, which is often lower due to an on-chip regulator (e.g., 1.2V).

Note that we also already have brownout-detection enabled, but that simply prevents causes a reset or makes the chip refuse to start. This can catch eventual brownout in advance, before the battery gets much too low.

I use the on-chip bandgap reference and compare it with a scaled version of the supplied voltage. The SAMD51 is very touchy about how to use this reference. See https://community.atmel.com/forum/samd51-using-intref-adc-voltage-reference.

- I did some code cleanup in the `atmel-same` `Processor.c` as well, to remove some confusion I introduced about ADC channels vs `MUXPOS` choices.
- The nRF chips have a simple way to calibrate the ADC, which is recommended on startup. I added this.
- I attempted to use the `INTREF` for the temperature measurement as well, based on new info about using it for the voltage, but this still does not work.

Tested using a variable bench power supply. The SAMD chips can clearly measure sag. On the nRF52840, I lost the USB connection very early when testing sag, but it still seems to work.

### CODE REVIEW

**Comment by ladyada** [comment]
yay! ill check next week i think

File: ports/nrf/common-hal/analogio/AnalogIn.c [hunk 1]
```diff
@@ -29,11 +29,21 @@
...
 #include "supervisor/shared/translate.h"
 
-#include "nrfx_saadc.h"
+#include "nrf_saadc.h"
 #include "nrf_gpio.h"
 
 #define CHANNEL_NO 0
 
+void analogin_reset(void) {
+    // Calibrate the ADC once, on startup.
```
Comment (tannewt) [inline ports/nrf/common-hal/analogio/AnalogIn.c]: _reset is called after each vm run. Do you want this in _init instead?

Comment (dhalbert) [reply ports/nrf/common-hal/analogio/AnalogIn.c]: That was unintentional, but the calibration can vary based on temperature, so maybe it's a good idea to do it on each vm run. It takes maybe a msec or so.

Comment (dhalbert) [reply ports/nrf/common-hal/analogio/AnalogIn.c]: Alternatively, we can call it once, and add a `calibrate()` function which is a no-op on other ports. I actually didn't notice any difference in the results before and after calibration, though.

Comment (dhalbert) [reply ports/nrf/common-hal/analogio/AnalogIn.c]: OK, I see some other calibration possibilities for other chips. I'll change it to `analogin_init()` and maybe add a `calibrate()` function later.

**Comment by dhalbert** [comment]
@tannewt all updated to handle recent PR's

**Pull request merged**